WordPress/Classes/ViewRelated/NUX/Post Signup Interstitial/PostSignUpInterstitialViewController.swift

```diff
@@ -46,6 +49,9 @@ class PostSignUpInterstitialViewController: UIViewController {
         view.backgroundColor = .listBackground
 
         configureI18N()
+
+        //Mark it as seen
+        PostSignUpInterstitialViewController.markAsSeen()
```

Comment:
This still reads very strangely to me.

If we aim to decouple the seen/not seen logic from the ViewController we can extract it to another entity. This entity might even be encapsulated inside this VC if you want to  (and it can be a struct, since there's no state at all).

I think my discomfort here comes from two things:

- `static`/`class` being used for the purpose of decoupling (which I believe it's not what static/class is intended for), you can see here that the static name has this class name, so they're related :P 
- It's very procedural
- We're breaking the single responsibility principle: the VC should only present data and here we're also handling a database / saved state.

This VC is quite simple, so I don't think it's a huge problem if we keep this at it is. Just my 2 cents for you to think about. :)

Reply:
After taking another look at this, this morning I agree. I'll take another pass at my implementation.